packages/server-wallet/src/config/validation.ts

```diff
@@ -0,0 +1,101 @@
+import joi, {ValidationErrorItem} from 'joi';
+import {parse} from 'pg-connection-string';
+
+import {ServerWalletConfig} from './types';
+
+const validateConnectionString = (connection: string) => {
+  const parsed = parse(connection);
+  const result = databaseObjectConfigurationSchema.validate(parsed);
+  if (result.error) {
+    throw new Error('Invalid connection string');
+  }
+  return parsed;
+};
+const databaseObjectConfigurationSchema = joi.object({
+  database: joi.string().required(),
+  host: joi.string().required(),
+  user: joi.string().required(),
+  password: joi
+    .string()
+    .allow('')
+    .optional(),
+  port: joi
+    .number()
+    .integer()
+    .optional(),
+});
+const databaseConnectionConfigurationSchema = joi.alternatives().conditional('.', {
+  is: joi.string(),
+  then: joi.string().custom(validateConnectionString, 'The connection string is valid'),
+  otherwise: databaseObjectConfigurationSchema,
+});
+
+const databasePoolConfigurationSchema = joi.object({
+  max: joi
+    .number()
+    .integer()
+    .min(0)
+    .optional(),
+  min: joi
+    .number()
+    .integer()
+    .min(0)
+    .optional(),
+});
+
+const databaseConfigurationSchema = joi.object({
+  connection: databaseConnectionConfigurationSchema.required(),
+  debug: joi.boolean().optional(),
+  pool: databasePoolConfigurationSchema.optional(),
+});
+
+const metricsConfigurationSchema = joi.object({
+  timingMetrics: joi.boolean().required(),
+  metricsOutputFile: joi.string().when('timingMetrics', {is: true, then: joi.string().required()}),
+});
+
+const loggingConfigurationSchema = joi.object({
+  logLevel: joi.string().valid('fatal', 'error', 'warn', 'info', 'debug', 'trace'),
+  logDestination: joi.string().required(),
+});
+
+const networkConfigurationSchema = joi.object({
+  rpcEndpoint: joi
+    .string()
+    .uri()
+    .optional(),
+  chainNetworkID: joi
+    .number()
+    .integer()
+    .min(0)
+
+    .required(),
+});
+
+const serverWalletConfigSchema = joi.object({
```

Comment:
It seems like we could use type parameters to improve maintainability here.

```suggestion
const serverWalletConfigSchema = joi.object<ServerWalletConfig>({
```